src/NUnitFramework/framework/Attributes/RangeAttribute.cs

```diff
@@ -218,7 +174,34 @@ public RangeAttribute(float from, float to, float step)
         /// <param name="parameter">The parameter of a parameterized test.</param>
         public IEnumerable GetData(Type fixtureType, ParameterInfo parameter)
         {
-            return ParamAttributeTypeConversions.ConvertData(_data, parameter.ParameterType);
+            var from = ParamAttributeTypeConversions.Convert(_from, parameter.ParameterType);
+            var to = ParamAttributeTypeConversions.Convert(_to, parameter.ParameterType);
+
+            var valueGenerator = ValueGenerator.Create(parameter.ParameterType);
+
+            ValueGenerator.Step step;
+            if (!valueGenerator.TryCreateStep(_step, out step))
+            {
+                // ValueGenerator.CreateStep is responsible to enable incrementing bytes by (int)(-1),
+                // or perhaps in the future a DateTime by a TimeSpan, but the responsibility to convert
+                // attribute values from Double to Decimal is in ParamAttributeTypeConversions.
+                // (ParamAttributeTypeConversions tries to simulate what would happen in C# if the
+                // literal syntax used for the attribute value had been used as a parameter argument
+                // in a direct call to the test method.)
+                object stepValueToRequire;
+                if (!ParamAttributeTypeConversions.TryConvert(_step, parameter.ParameterType, out stepValueToRequire))
+                    stepValueToRequire = _step;
```

Comment:
Yes. This will cause CreateStep to throw the same exception as it would throw if TryConvert succeeded but the value generator still didn’t recognize the step value. 📄